lib/engine/part/city.rb

```diff
@@ -6,14 +6,15 @@
 module Engine
   module Part
     class City < RevenueCenter
-      attr_accessor :reservations
+      attr_accessor :reservations, :block_if
```

Comment:
what is this for?

Reply:
Allow the game to add any conditions, beyond the token, that make a city blocked. For 1828, the VA coalfields tile is blocked unless the corporation has bought a coal marker.

Reply:
ok, let's not do this. since it's a one off thing for for 1828.

instead, just overwrite the whole function on g_1828#244

Reply:
What technique do you suggest? Overwrite the method dynamically in the instance? I ask because before I did the block_if approach, I went down the inheritance path. I had to overwrite city.rb, tile.rb, and init_tiles in the game base class. Seemed like a lot and the approach above felt like a more elegant way to inject behavior from the game.

Reply:
I think you should be able to dynamically reassign the block? method of this one city just like you assign block_if

Reply:
`irb(main):001:1* class X
irb(main):002:2*   def y
irb(main):003:2*     puts 'hi'
irb(main):004:1*   end
irb(main):005:0> end
=> :y
irb(main):006:0> x = X.new
irb(main):007:0> x.y
hi
=> nil
irb(main):008:1* def x.y
irb(main):009:1*   puts 'z'
irb(main):010:0> end
=> :y
irb(main):011:0> x.y
z

`

Reply:
Okay, sounds good. I'm familiar with that approach and surprised Opal can handle. Cool.